### PR TITLE
Feat/api keys

### PR DESCRIPTION
**Pull request by blackicon-eth**

### CODE REVIEW

File: app/api/public/users/[fid]/status/route.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  req: NextRequest,
+  {
+    params: { fid },
+  }: {
+    params: { fid: string };
+  }
+) => {
+  try {
+    // Get the API key from the headers
+    const apiKey = req.headers.get("x-api-key");
```
Comment (limone-eth) [inline app/api/public/users/[fid]/status/route.ts:14]: Let's create a function that:
- gets in a req: NextRequest and mode: string parameters
- performs all the checks that you are doing between line 14 and line 44
- returns a boolean
- before returning, if the boolean result is true (api-key and mode are valid for the current endpoint), create a request in the db as you are doing on line 47

File: app/api/public/users/route.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    // get the x-waitlist-id from the headers that surely exists because of the middleware
+    const waitlistId = key.waitlist_id;
+
+    // Find the waitlist associated with the id
+    const waitlist = await prisma.waitlist.findUnique({
+      where: {
+        id: waitlistId,
+      },
+      include: {
```
Comment (limone-eth) [inline app/api/public/users/route.ts:61]: Let's remove this _count check, as we (for now) allow only QUEEN waitlist to use the API, so no limit on users to add

File: components/UsersTable.tsx [hunk 3]
```diff
@@ -374,6 +397,34 @@ export const UsersTable = ({
...
                 {new Date(user.waitlistedAt).toDateString()}
               </TableCell>
+              <TableCell>
+                <Checkbox
+                  defaultSelected={
+                    user.status === WaitlistedUserStatus.APPROVED
+                  }
+                  onChange={async () => {
+                    // Change user status in the database with a PUT request
+                    await fetch(
```
Comment (limone-eth) [inline components/UsersTable.tsx:407]: Abstract this in a function

File: app/api/public/users/route.ts (hunk 2 above)
Comment (blackicon-eth) [reply app/api/public/users/route.ts:61]: Should we also disable the checkbox if the waitlist is not a QUEEN one?
Also: should we add a line in the readme and in the pricing page to state this new feature?

**Comment by limone-eth** [comment]
Yes, let’s do it

Simone Staffa
Founder @Shift
LinkedIn <https://www.linkedin.com/in/simone-staffa-8b3b79158/> | X
<https://x.com/limone_eth> | Book a meeting <https://cal.com/simone-staffa>

**Pull request merged**